client/validators/profile.ts

```diff
@@ -85,3 +90,39 @@ export const formProfile = ({field, value, profile, errors, messages, diff}) =>
         delete errors[field];
     }
 };
+
+interface IValidateCustomCVArgs {
+    field: string;
+    value: undefined; // Value is `undefined`, as there is no field called `custom_vocabularies`
+    profile: IEditorProfile;
+    errors: {[field: string]: string};
+    messages: Array<string>;
+    diff: Partial<IEventOrPlanningItem>;
+}
+
+export function formProfileCustomVocabularies({field, value, profile, errors, messages, diff}: IValidateCustomCVArgs) {
+    const schema = profile.schema?.[field];
+
+    if (schema == null || schema.type !== 'list' || profile.editor[field]?.enabled !== true) {
+        return;
+    } else if (schema.required !== true) {
+        return;
+    }
+
+    planningApi.vocabularies.getCustomVocabularies()
+        .filter((cv) => schema.vocabularies?.includes(cv._id))
```

Comment:
`.filter((cv) => (schema.vocabularies ?? []).includes(cv._id))`

a default would be better here - it's easier to quickly verify that output will be a boolean we won't get surprises by returning `undefined` from callback function